driver-core/src/main/com/mongodb/internal/operation/CreateCollectionOperation.java

```diff
@@ -361,6 +362,27 @@ private BsonDocument getCreateCollectionCommand() {
         return document;
     }
 
+    private void checkEncryptedFieldsSupported(final ConnectionDescription connectionDescription) throws MongoException {
+        if (encryptedFields != null && serverIsLessThanVersionSevenDotZero(connectionDescription)) {
+            throw new MongoClientException("Driver support of Queryable Encryption is incompatible with server."
+                    + " Upgrade server to use Queryable Encryption.");
+        }
+    }
+
+    /**
+     * @return {@code true} iff the {@linkplain #checkEncryptedFieldsSupported(ConnectionDescription) check} was successful,
```

Comment:
Is this a new async design pattern you're introducing?  It doesn't look familiar.

Reply:
The same approach is used by `LoopState.breakAndCompleteIf`/`RetryState.breakAndCompleteIfRetryAnd` to make async callback-based code to look similar to synchronous code.